### PR TITLE
Add version detection to BrowserDetection

### PR DESCRIPTION
**Pull request by danimoh**

### CODE REVIEW

File: src/browser-detection/BrowserDetection.ts [hunk 1]
```diff
@@ -45,6 +54,47 @@ class BrowserDetection {
...
+                regex = /(Opera|OPR)\/(\S+)/i;
+                break;
+            case BrowserDetection.Browser.FIREFOX:
+                regex = /Firefox\/(\S+)/i;
+                break;
+            case BrowserDetection.Browser.CHROME:
+                regex = /Chrome\/(\S+)/i;
+                break;
+            case BrowserDetection.Browser.SAFARI:
+                regex = /Version\/(\S+)/i;
```
Comment (Bettelstab) [inline src/browser-detection/BrowserDetection.ts]: Are you sure this is sufficient for iOS Safari? In my experience, minor versions make a great difference there and in the example above, we need 11.4.1, don't we?

Comment (danimoh) [reply src/browser-detection/BrowserDetection.ts]: No, not sure ;)
What does apple's iOS release versioning policy look like?
Are new Safari versions always released along new os versions with the same version number?
Note that there is also / still the method `iOSversion` you implemented back then.
Should I return the version retrieved from that method for mobile safari?
The `Version/*` part only seams to return the major version.

Comment (Bettelstab) [reply src/browser-detection/BrowserDetection.ts]: AFAIK iOS Safari version is bound to OS version, so yes ;)

Comment (danimoh) [reply src/browser-detection/BrowserDetection.ts]: Please check whether the newest changes are ok for you.

**Pull request merged**